packages/block-library/src/navigation/edit/index.js

```diff
@@ -224,36 +224,54 @@ function Navigation( {
 	// than 1 exists then use the most recent.
 	// The aim is for the block to "just work" from a user perspective using existing data.
 	useEffect( () => {
-		if (
-			hasUncontrolledInnerBlocks ||
-			isCreatingNavigationMenu ||
-			ref ||
-			! navigationMenus?.length
-		) {
+		// Only autofallback to published menus.
+		const fallbackNavigationMenus = navigationMenus?.filter(
+			( menu ) => menu.status === 'publish'
+		);
+
+		if ( hasUncontrolledInnerBlocks || isCreatingNavigationMenu || ref ) {
 			return;
 		}
 
-		navigationMenus.sort( ( menuA, menuB ) => {
-			const menuADate = new Date( menuA.date );
-			const menuBDate = new Date( menuB.date );
-			return menuADate.getTime() < menuBDate.getTime();
-		} );
+		if ( ! fallbackNavigationMenus?.length && classicMenus?.length === 1 ) {
+			// If there's only one classic menu
+			// then create a new navigation menu based on it.
 
-		// Only autofallback to published menus.
-		const fallbackNavigationMenus = navigationMenus.filter(
-			( menu ) => menu.status === 'publish'
-		);
-		if ( fallbackNavigationMenus.length === 0 ) return;
+			// This is duplicated several times.
+			const onSelectClassicMenu = async ( classicMenu ) => {
+				const navMenu = await convertClassicMenu(
+					classicMenu.id,
+					classicMenu.name,
+					'publish'
+				);
+				if ( navMenu ) {
+					handleUpdateMenu( navMenu.id, {
+						focusNavigationBlock: true,
+					} );
+				}
+			};
+
+			onSelectClassicMenu( classicMenus[ 0 ] );
```

Comment:
I think here we should use the same process that we use for the block menus, that is avoid calling `handleUpdateMenu`. This call sets the ref and the use sees a dirty post. We should instead just _create_ the block menu, then _because it's the latest created_ it will be handled by the later code.